### PR TITLE
Update errorprone

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: api/all/src/main/java/io/opentelemetry/api/common/Attributes.java [hunk 1]
```diff
@@ -140,6 +140,7 @@ static Attributes empty() {
    * Returns a {@link Attributes} instance with the given key-value pairs. Order of the keys is not
    * preserved. Duplicate keys will be removed.
    */
+  @SuppressWarnings("TooManyParameters")
```
Comment (jkwatson) [inline api/all/src/main/java/io/opentelemetry/api/common/Attributes.java:143]: lol.

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedOutputStream.java [hunk 2]
```diff
@@ -62,7 +62,7 @@
 // - No support for writing fields with tag, we alway write tags separately
 // - Allow resetting and use a ThreadLocal instance
 //
-@SuppressWarnings({"UngroupedOverloads", "InlineMeSuggester"})
+@SuppressWarnings({"UnnecessaryFinal", "UngroupedOverloads", "InlineMeSuggester"})
```
Comment (jkwatson) [inline exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedOutputStream.java:65]: why is this needed? Oh.... is this for final variables in the code?

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedInputStream.java [hunk 3]
```diff
@@ -49,6 +49,7 @@
  * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
  * at any time.
  */
+@SuppressWarnings("UnnecessaryFinal")
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedInputStream.java:52]: Is the `final` unnecessary because it has a private constructor?

**Review by jack-berg — APPROVED** [review]
Big fan of getting rid of the `final` keyword in methods where the var is already effectively final 👍

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorageTest.java [hunk 4]
```diff
@@ -6,7 +6,6 @@
 package io.opentelemetry.sdk.metrics.internal.state;
 
 import static io.opentelemetry.sdk.testing.assertj.MetricAssertions.assertThat;
-import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
```
Comment (jkwatson) [inline sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorageTest.java:9]: how was this not just a straight-up compiler error?

**Review by jkwatson — APPROVED** [review]
❤️  getting rid of all the un-needed `final` noise!

**Comment by Oberon00** [comment]
Coming from C++, I view getting rid of final as bad. Final (like const in C/C++) means that you get a compiler error if you assign to the variable by accident.

**Comment by jkwatson** [comment]
> Coming from C++, I view getting rid of final as bad. Final (like const in C/C++) means that you get a compiler error if you assign to the variable by accident. (Just adding this as a tidbit, don't view this as a vote for or against anything)

Sure...but if your methods are nice and tight and small, this shouldn't be an issue! It should be easy to see by inspection that you might be re-assigning a parameter. Also, much less of an issue when there are no pointers passed around to have caller side-effects in Java. :)

File: exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedInputStream.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (anuraaga) [reply exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/CodedInputStream.java:52]: These two classes are mostly copied as is don't think we should drift them too much from upstream just for style checks
